plugin-cgroupv2/src/k8s/utils.rs

```diff
@@ -468,37 +607,736 @@ mod tests {
+                        {
+                            "metadata": {
+                                "name": "pod2",
+                                "namespace": "default",
+                                "uid": "56789"
+                            },
+                            "spec": {
+                                "nodeName": "node2"
+                            }
+                        }
+                    ]
+                })
+                .to_string(),
+            )
+            .create();
+
+        let kubernetes_api_url = &mockito::server_url();
+        let mut token = Token::new(TokenRetrieval::File);
+
+        token.path =
+            Some("/tmp/test-alumet-plugin-k8s/var_4/run/secrets/kubernetes.io/serviceaccount/token_4".to_string());
+
+        let result = get_existing_pods(node, kubernetes_api_url, &token).await.unwrap();
+
+        assert_eq!(
+            result.get("hash1").unwrap(),
+            &("pod1".to_string(), "default".to_string(), "node1".to_string())
+        );
+        assert_eq!(
+            result.get("56789").unwrap(),
+            &("pod2".to_string(), "default".to_string(), "node2".to_string())
+        );
+
+        std::fs::remove_dir_all(&root).unwrap();
+    }
+
+    // Test `get_existing_pods` with JSON send in fake server to a specific token,
+    // with some of them missing in the JSON
+    #[tokio::test]
+    async fn test_get_existing_pods_with_half_valid_data() {
+        let tmp = std::env::temp_dir();
+        let root = tmp.join("test-alumet-plugin-k8s/var_5/");
+
+        if root.exists() {
+            std::fs::remove_dir_all(&root).unwrap();
+        }
+
+        let dir = root.join("run/secrets/kubernetes.io/serviceaccount/");
+        std::fs::create_dir_all(&dir).unwrap();
+
+        let path = dir.join("token_5");
+        std::fs::write(&path, TOKEN_CONTENT).unwrap();
+
+        let node = "pod1";
+        let url = format!("/api/v1/pods/?fieldSelector=spec.nodeName={}", node);
+        let _mock = mock("GET", url.as_str())
+            .with_status(200)
+            .with_header("Content-Type", "application/json")
+            .with_body(
+                json!({
+                    "items": [
+                        {
+                            "metadata": {
+                                "namespace": "default",
+                                "annotations": {
+                                    "kubernetes.io/config.hash": "hash1"
+                                }
+                            },
+                            "nodeName": "node1",
+                        },
+                    ]
+                })
+                .to_string(),
+            )
+            .create();
+
+        let kubernetes_api_url = &mockito::server_url();
+        let mut token = Token::new(TokenRetrieval::File);
+
+        token.path =
+            Some("/tmp/test-alumet-plugin-k8s/var_5/run/secrets/kubernetes.io/serviceaccount/token_5".to_string());
```

Comment:
Don't hardcode the path, and use `tempfile::tempdir` to get a new unique temporary directory.